test/unit/messaging/process-statement-message.test.js

```diff
@@ -1,23 +1,88 @@
 jest.mock('ffc-messaging')
+jest.mock('../../../app/data')
+const mockGenerator = jest.fn()
+jest.mock('../../../app/generator', () => {
+  return mockGenerator
+})
+const mockValidation = jest.fn()
+jest.mock('../../../app/messaging/validate-statement', () => {
+  return mockValidation
+})
 const processStatementMessage = require('../../../app/messaging/process-statement-message')
+const mockStatement = require('../../mock-statement-data')
+const { VALIDATION } = require('../../../app/errors')
 let receiver
 
 describe('process statement message', () => {
   beforeEach(() => {
+    jest.resetAllMocks()
     receiver = {
-      completeMessage: jest.fn()
+      completeMessage: jest.fn(),
+      deadLetterMessage: jest.fn()
     }
   })
 
-  afterEach(() => {
-    jest.clearAllMocks()
+  test('completes message on success', async () => {
+    const message = {
+      body: mockStatement
+    }
+    await processStatementMessage(message, receiver)
+    expect(receiver.completeMessage).toHaveBeenCalledWith(message)
+    expect(receiver.completeMessage).toHaveBeenCalledTimes(1)
   })
 
-  test('completes message', async () => {
+  test('calls validate statement', async () => {
     const message = {
-      body: {}
+      body: mockStatement
     }
     await processStatementMessage(message, receiver)
-    expect(receiver.completeMessage).toHaveBeenCalledWith(message)
+    expect(mockValidation).toHaveBeenCalledWith(message.body)
+    expect(mockValidation).toHaveBeenCalledTimes(1)
+  })
+
+  test('calls generator with statement', async () => {
+    const message = {
+      body: mockStatement
+    }
+    await processStatementMessage(message, receiver)
+    expect(mockGenerator).toHaveBeenCalledWith(message.body)
+    expect(mockGenerator).toHaveBeenCalledTimes(1)
+  })
+
+  test('does not complete message on error', async () => {
+    const message = {
+      body: mockStatement
+    }
+    mockGenerator.mockImplementation(() => { throw new Error('Unable to generate') })
+    await processStatementMessage(message, receiver)
+    expect(receiver.completeMessage).not.toHaveBeenCalledWith(message)
+    expect(receiver.deadLetterMessage).not.toHaveBeenCalledWith(message)
```

Comment:
think this needs tweaking to: `expect(receiver.completeMessage).not.toHaveBeenCalled()`

the existing test would pass if both of these funcs were called w/ anything other than `message`, however we dnt want them to be called at all which the tweaked test handles

Reply:
Yes true.  Fixed